Nos/Service/Relay/RelayService.swift

```diff
@@ -731,10 +733,12 @@ extension RelayService {
             
             // Try to publish each of these again to each relay that failed.
             for event in eventsToRetry {
+                guard let jsonEvent = event.codable else { continue }
+                
                 let missedRelays = event.shouldBePublishedTo.subtracting(event.publishedTo)
+                let missedAddresses = missedRelays.compactMap { $0.address }
                 
-                for missedRelay in missedRelays {
-                    guard let missedAddress = missedRelay.address, let jsonEvent = event.codable else { continue }
+                for missedAddress in missedAddresses {
```

Comment:
I have two different feelings regarding this. On the one hand, the `for` looks easier to read, on the other hand, it is less computationally efficient. We are lopping through missedRelays and then looping through missedAddress.

What about a middle ground? That would mean 1) keep the movement  of `guard let jsonEvent = event.codable else { continue }` to the top as it is, but 2) keep doing `for missedRelay in missedRelays { guard let missedAddress = missedRelay.address else { continue } ... }` (without the `jsonEvent` clause).

Reply:
This is all just my opinion, so I'd like to leave it as is, but...
Since the array of missedRelays is always going to be relatively small (how many relays can a user have at maximum?), it is a worthwhile tradeoff to make for readability. And generally I consider a `guard` at the top of the body of a loop to be a code smell because there are other mechanisms for filtering a collection, such as a `where` clause in your for loop (`for thing in things where thing.someCondition`). Optimizing for performance here is premature in my opinion.